_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,298 @@
+file.  For example:
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: details
+  labels:
+    app: details
+  namespace: ns-1
+```
+Although not shown, the application namespaces will also have RBAC settings limiting access
+to certain resources. These RBAC settings could be set by the cluster administator and/or
+the tenant administator.
+
+#### Using Istioctl commands in a multitenant environment
+When defining route rules (`RouteRule`) or destination policies (`DestinationPolicy`), it is
+necessary to ensure that the `istioctl` command is scoped to
+the `namespace` the Istio control plane is running in to ensure the resource is created
+in the proper namespace. Additionally, the rule itself must be scoped to the tenant's namespace
+so that it will be applied properly to that tenant's mesh.  The "-i" option is used to create
+(or get or describe) the rule in the namespace that the istio control plane is deployed in.
+The "-n" option will scope the rule to the tenant's mesh and should be set to the namespace that
+the tenant's app is deployed in. Note that the -n option can be skipped on the command line if
+the .yaml file for the resource scopes it properly instead.
+
+For example, the following command would be required to add a route rule to the *istio-system1*
+namespace:
+```sh
+istioctl –i istio-system1 create -n ns-1 -f route_rule_v2.yaml
+```
+
+```sh
+istioctl -i istio-system1 -n ns-1 get routerule
+NAME			KIND					NAMESPACE
+details-Default		RouteRule.v1alpha2.config.istio.io	ns-1
+productpage-default	RouteRule.v1alpha2.config.istio.io	ns-1
+ratings-default		RouteRule.v1alpha2.config.istio.io	ns-1
+reviews-default		RouteRule.v1alpha2.config.istio.io	ns-1
+```
+
+#### Test Results
+Following the instructions above, a cluster administrator can create an environment limiting
+(via RBAC and namespaces) what a tenant administrator can deploy. 
+
+After deployment, accessing the Istio control plane pods assigned to a specific tenant
+administrator is permitted:
+```sh
+dev-admin:~/go/src/istio.io/istio# kubectl get pods -n istio-system
+NAME                                      READY     STATUS    RESTARTS   AGE
+grafana-78d649479f-8pqk9                  1/1       Running   0          1d
+istio-ca-ffbb75c6f-98w6x                  1/1       Running   0          1d
+istio-ingress-68d65fc5c6-dnvfl            1/1       Running   0          1d
+istio-mixer-5b9f8dffb5-8875r              3/3       Running   0          1d
+istio-pilot-678fc976c8-b8tv6              2/2       Running   0          1d
+istio-sidecar-injector-7587bd559d-5tgk6   1/1       Running   0          1d
+prometheus-cf8456855-hdcq7                1/1       Running   0          1d
+servicegraph-75ff8f7c95-wcjs7             1/1       Running   0          1d
+```
+However, accessing all the cluster's pods or pods from another tenant's namespace is not:
+```sh
+dev-admin:~/go/src/istio.io/istio# kubectl get pods --all-namespaces
```

Comment:
Separate command-line from output in two blocks.
Don't include the prompt in the command-line.